packages/postcss-lowercase-text/src/__tests__/index.test.js

```diff
@@ -0,0 +1,156 @@
+import postcss from 'postcss';
+
+import plugin from '..';
+
+function run(testMsg = 'should work', input, output) {
+  it(testMsg, (done) => {
+    let result = postcss([plugin]).process(input, {
+      from: undefined,
+    });
+    expect(result.css).toEqual(output);
+    done();
+  });
+}
+
+describe('transforming CSS properties', () => {
+  run(
+    'should safely transform the  CSS  Properties to lowercase : BORDER',
+    '#main{BORDER: 1px solid black;}',
+    '#main{border: 1px solid black;}'
+  );
+  run(
+    'should safely transform the  CSS  Properties to lowercase : BACKGROUND-COLOR',
+    '#main{BACKGROUND-COLOR: black;}',
+    '#main{background-color: black;}'
+  );
+});
+
+describe('transforming CSS selectors', () => {
+  run(
+    'should safely transform the  CSS id Selectors to lowercase ',
+    '#main{border: 1px solid black;}',
+    '#main{border: 1px solid black;}'
+  );
+
+  run(
+    'should safely transform the  CSS class Selectors to lowercase',
+    '.main{border: 1px solid black;}',
+    '.main{border: 1px solid black;}'
+  );
+
+  run(
+    'should safely transform the CSS HTML tag Selectors to lowercase',
+    'LI{border: 1px solid black;}',
+    'li{border: 1px solid black;}'
+  );
+
+  run(
+    'no transform to lowercase required',
+    'li{border: 1px solid black;}',
+    'li{border: 1px solid black;}'
+  );
+
+  run(
+    'should safely transform the  CSS caps multiple HTML Selectors to lowercase',
+    'UL LI{border: 1px solid black;}',
+    'ul li{border: 1px solid black;}'
+  );
+
+  run(
+    'should safely transform the  CSS HTML Selectors to lowercase',
+    'UL LI, p A{border: 1px solid black;}',
+    'ul li, p a{border: 1px solid black;}'
+  );
+
+  run(
+    'should safely transform the  CSS attribute Selectors to lowercase',
+    '[NAME="newname"]{border: 1px solid black;}',
+    '[name="newname"]{border: 1px solid black;}'
+  );
```

Comment:
Please add test: `LI[DATA-VALUE|="1900"]`, we should not lowercase `DATA-VALUE` (keep as is) because https://html.spec.whatwg.org/multipage/dom.html#embedding-custom-non-visible-data-with-the-data-*-attributes